packages/webpack-cli/lib/groups/resolveOutput.js

```diff
@@ -0,0 +1,21 @@
+const path = require('path');
+
+/**
+ * Resolves the output flag
+ * @param {args} args - Parsed arguments passed to the CLI
+ */
+const resolveOutput = (args) => {
+    const { output } = args;
+    const finalOptions = {
+        options: { output: {} },
+        outputOptions: {},
+    };
+    if (output) {
+        const { dir, base } = path.parse(output);
+        finalOptions.options.output.path = path.resolve(dir);
```

Comment:
this is because `dir` can be both an absolute or relative route

Reply:
To be honestly we have a lot of flags where we can use path, especial in webpack@5, can you open new issue about it, for example `--cache-location`

Reply:
here https://github.com/webpack/webpack-cli/issues/1841